tests/unit/Option/NoneTest.php

```diff
@@ -84,6 +85,19 @@ public function testContains(): void
         static::assertFalse($option->contains(4));
     }
 
+    public function testApply(): void
+    {
+        $spy = new Ref(1);
+
+        $option = Option\none();
+        $option->apply(static function (int $value) use ($spy) {
```

Comment:
You are not validating the returned result at the moment:

```php
$actual = $option->apply(...);

static::assertSame($option, $actual);
```

Reply:
I would say that this line would be the same assertion if I'm not mistaken: https://github.com/azjezz/psl/pull/426/files/89ed717b7d6801c77710f697b54a25ea1a0f959d#diff-1b95ed4d6bb75bfe486ce416a6a87705dc3dfffa381712a06f065d0425c9c749R98

Do you want it to be replaced by your suggestion?

Reply:
No problem :)
The some() version requires a similar approach though.